gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckExplicitSourceCompatibilityTask.java

```diff
@@ -92,12 +95,7 @@ public final void setShouldFix(boolean value) {
 
     @TaskAction
     public final void taskAction() throws IOException {
-        // We're doing this naughty casting because we need access to the `getRawSourceCompatibility` method.
-        org.gradle.api.plugins.internal.DefaultJavaPluginConvention convention =
```

Comment:
The internal API got moved to a different class. Naively, we could fix this with reflection based on the runtime Gradle version but that doesn't sound sustainable. Maybe this is a good time to lean into toolchains?

Reply:
doesn't java plugin extension exist since gradle 4.10? Maybe we can standardise on that?

Reply:
The extension exists since then but the internal method `getRawSourceCompatibility` that we abuse here got moved from the convention to the extension with Gradle 7. So we can't use the extension for versions before that.

I'll do a bit more digging. Still hoping we find a non-internal way of figuring out if sourceCompat is explicitly set. Another alternative would be to actually look into projects gradle files and fail of sourceCompat is not set, similar to the baseline error-prone rules.